nipype/pipeline/plugins/multiproc.py

```
@@ -100,3 +177,126 @@ def _report_crash(self, node, result=None):
+
+        # Check available system resources by summing all threads and memory used
+        busy_memory_gb = 0
+        busy_processors = 0
+        for jobid in jobids:
+            busy_memory_gb += self.procs[jobid]._interface.estimated_memory_gb
+            busy_processors += self.procs[jobid]._interface.num_threads
+
+        free_memory_gb = self.memory_gb - busy_memory_gb
+        free_processors = self.processors - busy_processors
+
+        # Check all jobs without dependency not run
+        jobids = np.flatnonzero((self.proc_done == False) & \
+                                (self.depidx.sum(axis=0) == 0).__array__())
+
+        # Sort jobs ready to run first by memory and then by number of threads
+        # The most resource consuming jobs run first
+        jobids = sorted(jobids,
+                        key=lambda item: (self.procs[item]._interface.estimated_memory_gb,
+                                          self.procs[item]._interface.num_threads))
+
+        logger.debug('Free memory (GB): %d, Free processors: %d',
+                     free_memory_gb, free_processors)
+
+        # While have enough memory and processors for first job
+        # Submit first job on the list
+        for jobid in jobids:
+            logger.debug('Next Job: %d, memory (GB): %d, threads: %d' \
+                         % (jobid, self.procs[jobid]._interface.estimated_memory_gb,
+                            self.procs[jobid]._interface.num_threads))
+
+            if self.procs[jobid]._interface.estimated_memory_gb <= free_memory_gb and \
+               self.procs[jobid]._interface.num_threads <= free_processors:
+                logger.info('Executing: %s ID: %d' %(self.procs[jobid]._id, jobid))
+                executing_now.append(self.procs[jobid])
+
+                if isinstance(self.procs[jobid], MapNode):
+                    try:
+                        num_subnodes = self.procs[jobid].num_subnodes()
+                    except Exception:
```

So maybe reraise the exception and add move the cleanup to a "finally" clause?